src/main/java/com/microsoft/sqlserver/jdbc/DDC.java

```diff
@@ -72,6 +72,17 @@ static final Object convertIntegerToObject(int intValue, int valueLength, JDBCTy
                 return (float) intValue;
             case BINARY:
                 return convertIntToBytes(intValue, valueLength);
+            case SQL_VARIANT:
+                // return short or bit if the underlying datatype of sql_variant is tinyint, smallint or bit
+                // otherwise, return integer
+                // Longer datatypes such as double and float are handled by convertLongToObject instead.
+                if (valueLength == 1) {
+                    return 0 != intValue;
+                } else if (valueLength == 3 || valueLength == 4) {
```

Comment:
can you add a comment here on why these 2 values?

Reply:
I do have a comment, on line 76